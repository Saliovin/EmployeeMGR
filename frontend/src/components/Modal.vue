<script setup lang="ts">
import Icon from './Icon.vue';

interface Props {
    title?: string;
}

const props = defineProps<Props>();
const emit = defineEmits(["close"]);
</script>

<template>
    <Teleport to="body">
        <div class="modal" @click="emit('close')">
            <div class="modal-main" @click.stop>
                <div class="modal-top">
                    <h2 v-if="props.title">{{ props.title }}</h2>
                    <Icon icon="x" size="lg" @click="emit('close')" clickable class="close" />
                </div>
                <slot />
            </div>
        </div>
    </Teleport>
</template>

<style scoped>
.modal {
    position: fixed;
    display: flex;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(49, 55, 60, 0.6);
    place-items: center;
}

.modal-main {
    margin: auto;
    min-width: 25rem;
    max-width: 50%;
    background-color: #fff;
    padding: 1rem;
}

.modal-top {
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin-bottom: 1rem;
}

.close {
    transition-duration: 0.1s;
}

.close:hover {
    cursor: pointer;
    transform: scale(1.7);
}
</style>
