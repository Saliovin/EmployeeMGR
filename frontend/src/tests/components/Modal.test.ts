import Modal from '@/components/Modal.vue'
import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

describe('modal', () => {
    it('base', () => {
        const modal = mount(Modal, {
            props: {
                title: 'test',
            }
        })
        expect(modal.element).toMatchSnapshot()
    })
})