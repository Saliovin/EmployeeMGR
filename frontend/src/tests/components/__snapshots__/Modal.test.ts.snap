// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`modal > base 1`] = `<!--teleport start-->`;
